README.md

```diff
@@ -187,6 +187,14 @@ GitHub github = Feign.builder()
                      .target(GitHub.class, "https://api.github.com");
 ```
 
+If you need to pre-process the response before give it to the Decoder, you can use the `mapAndDecode` builder method:
+
+```java
+GitHub github = Feign.builder()
+                     .mapAndDecode((response, type) -> myTransformingLogic(response, type), new GsonDecoder())
```

Comment:
myTransformingLogic is coolio, but there's probably a better hint. Ex what made you want this in the first place? is there a concrete, but easy to understand example? Notably, the map happens *before* the decoding, so something like unwrapping a nested json envelope doesn't seem to be realistic.

Let's make this easy to understand!

ps.. leave out the github thing as their api is already nice :)